### PR TITLE
[#10920] Add testing for sortCoursesBy method in instructor home page.

### PR DESCRIPTION
**Pull request by YvetteJian**
Part of https://github.com/TEAMMATES/teammates/issues/10920

Outline of Solution

Add test case for sortCoursesBy()

### CODE REVIEW

**Comment by YvetteJian** [comment]
Hi, I did not modified the backend code, however the backend test fail and I have no idea why this happened. Could somebody help me? Thank you !

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts [hunk 1]
```diff
@@ -271,6 +280,26 @@ describe('InstructorHomePageComponent', () => {
             .feedbackSession.feedbackSessionName).toEqual('First Session');
   });
 
+  it('should sort courseTabModels by courseId in ascending order', () => {
+    const CopytActiveCourseTabModels: CourseTabModel[] = activeCourseTabModels;
```
Comment (domlimm) [inline src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts:284]:
```suggestion
    const activeCourses: CourseTabModel[] = activeCourseTabModels;
```

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts [hunk 2]
```diff
@@ -271,6 +280,26 @@ describe('InstructorHomePageComponent', () => {
...
+    CopytActiveCourseTabModels.push({
```
Comment (domlimm) [inline src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts:285]:
```suggestion
    activeCourses.push({
```

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts [hunk 3]
```diff
@@ -271,6 +280,26 @@ describe('InstructorHomePageComponent', () => {
...
+      sessionsTableRowModels: [],
+      sessionsTableRowModelsSortBy: SortBy.NONE,
+      sessionsTableRowModelsSortOrder: SortOrder.ASC,
+
+      hasPopulated: false,
+      isAjaxSuccess: true,
+      isTabExpanded: true,
+      hasLoadingFailed: false,
+    });
+    component.courseTabModels = CopytActiveCourseTabModels;
```
Comment (domlimm) [inline src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts:297]:
```suggestion
    component.courseTabModels = activeCourses;
```

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts [hunk 4]
```diff
@@ -271,6 +280,26 @@ describe('InstructorHomePageComponent', () => {
...
+      instructorPrivilege: testInstructorPrivilege,
+      sessionsTableRowModels: [],
+      sessionsTableRowModelsSortBy: SortBy.NONE,
+      sessionsTableRowModelsSortOrder: SortOrder.ASC,
+
+      hasPopulated: false,
+      isAjaxSuccess: true,
+      isTabExpanded: true,
+      hasLoadingFailed: false,
+    });
```
Comment (domlimm) [inline src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts:296]: Nit: To follow how the other tests are structured.
```suggestion
    });

```

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts [hunk 5]
```diff
@@ -271,6 +280,26 @@ describe('InstructorHomePageComponent', () => {
...
+    component.courseTabModels = CopytActiveCourseTabModels;
+    component.sortCoursesBy(SortBy.COURSE_ID);
```
Comment (domlimm) [inline src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts:298]: Nit: To follow how the other tests are structured.
```suggestion
    component.sortCoursesBy(SortBy.COURSE_ID);

```

**Review by domlimm — COMMENTED** [review]
Looks great! Some changes to be made on code formatting.

Forgot to add:
@YvetteJian Let's change `courseTabModels` at Line 283 to `courses`.

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts (hunk 1 above)
Comment (domlimm) [reply src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts:284]: Let's change `courseTabModels` at Line 283 to `courses`.

**Review by domlimm — APPROVED** [review]
LGTM! Thank you for your contribution!

**Review by weiquu — APPROVED** [review]
LGTM!

**Pull request merged**